compiler/src/dotty/tools/dotc/typer/RefChecks.scala

```diff
@@ -323,6 +323,14 @@ object RefChecks {
             overrideErrorMsg("no longer has compatible type"),
             (if (member.owner == clazz) member else clazz).srcPos))
 
+      /** Do types of term members `member` and `other` as seen from `self` match?
+       *  If not we treat them as not a real override and don't issue certain
+       *  error messages. Also, bridges are not generated in this case.
```

Comment:
Why not skip all error messages instead of only certain one? Currently it looks like we still emit overrideAccessError and overrideTargetNameError but if these methods are not true override then I don't see why we need to.

Reply:
In fact it's just the comment that is wrong. All errors go to overrideError which does the check whether it's a true override.

Reply:
> All errors go to overrideError which does the check whether it's a true override

In that case, why not immediately check for false overrides at the top of checkOverride before doing any of the checks?

Reply:
Because the false overrides check is expensive and override checks are common while override errors are rare.